# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/content/widgets/browser-custom-element.js
# Commit: f95c19e45ab2
# Full Hash: f95c19e45ab2ef6a1ba8ac8d4124df117928b1a4
# Author: Jonathan Watt <jwatt@jwatt.org>
# Date: 2020-07-22 21:55:45
# Regressor Bug: 1652270
# File Overlap Count: 1
# Description:
#   Bug 1652270. Convert nsFrameLoader::Print to return a Promise. r=farre,remote-protocol-reviewers,marionette-reviewers,jgraham,whimboo,mixedpuppy
#   
#   (Instead of requiring callers to pass an nsIWebProgressListener.)
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D83249
# ==============================================================================

diff -r c749c42dc6b9 -r f95c19e45ab2 toolkit/content/widgets/browser-custom-element.js
--- a/toolkit/content/widgets/browser-custom-element.js	Wed Jul 22 18:40:32 2020 +0300
+++ b/toolkit/content/widgets/browser-custom-element.js	Wed Jul 22 15:54:01 2020 +0000
@@ -1710,16 +1710,12 @@
       };
     }
 
-    print(aOuterWindowID, aPrintSettings, aPrintProgressListener) {
+    print(aOuterWindowID, aPrintSettings) {
       if (!this.frameLoader) {
         throw Components.Exception("No frame loader.", Cr.NS_ERROR_FAILURE);
       }
 
-      this.frameLoader.print(
-        aOuterWindowID,
-        aPrintSettings,
-        aPrintProgressListener
-      );
+      return this.frameLoader.print(aOuterWindowID, aPrintSettings);
     }
 
     async drawSnapshot(x, y, w, h, scale, backgroundColor) {
